File: packages/types/src/document.ts
export interface Document {
  id: string
  title: string
  content: string
  contentType: 'markdown' | 'pdf' | 'docx' | 'txt' | 'html'
  projectId?: string
  userId: string
  tags: string[]
  version: number
  fileUrl?: string
  fileSize?: number
  sourceUrl?: string
  summary?: string
  createdAt: string
  updatedAt: string
}

export interface DocumentChunk {
  id: string
  documentId: string
  content: string
  embedding: number[]
  metadata: Record<string, unknown>
  pageNumber?: number
  chunkIndex: number
}

export interface DocumentTemplate {
  id: string
  name: string
  description: string
  content: string
  category: 'report' | 'letter' | 'note' | 'article' | 'custom'
  createdAt: string
}

File: services/api/app/schemas/document.py
from pydantic import BaseModel
from datetime import datetime


class DocumentCreate(BaseModel):
    title: str
    content: str = ""
    content_type: str = "markdown"
    project_id: str | None = None
    tags: list[str] = []


class DocumentResponse(BaseModel):
    id: str
    title: str
    content: str
    content_type: str
    project_id: str | None = None
    tags: list[str] = []
    version: int = 1
    file_url: str | None = None
    file_size: int | None = None
    summary: str | None = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
